### PR TITLE
Rename RoutingSource to NavModel

### PR DESCRIPTION
**Pull request by zsoltk**
## Description

**Breaking change**: `RoutingSource` renamed to `NavModel`. All subclasses, fields, package names, etc., any mentions of the word follow suit.

Package and model renames:
- `com.bumble.appyx.routingsource` → `com.bumble.appyx.navmodel`
- `com.bumble.appyx.core.routing` → `com.bumble.appyx.core.navigation`
- `com.bumble.appyx.core.routing.source` → `com.bumble.appyx.core.navigation.model`
- `routing-source-addons` module → `navmodel-addons`

## Check list

- [x] I have updated `CHANGELOG.md` if required.
- [x] I have updated documentation if required.

### CODE REVIEW

File: documentation/navmodel/index.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+Navigation models are one of the most important abstractions of Appyx.
+
+## What's a NavModel?
+
+Navigation models describe navigation itself – by the states and operations they define, any custom navigation mechanism can be implemented. The state is then yours to be controlled from your business logic, which then becomes the state of navigation directly.
+
+`NavModel` capabilities differ from implementation to implementation, however, typically:
+
+1. They store information on the states all children
```
Comment (KovalevAndrey) [inline documentation/navmodel/index.md:11]:
```suggestion
1. They store information of the children states
```

File: documentation/navmodel/index.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+## What's a NavModel?
+
+Navigation models describe navigation itself – by the states and operations they define, any custom navigation mechanism can be implemented. The state is then yours to be controlled from your business logic, which then becomes the state of navigation directly.
+
+`NavModel` capabilities differ from implementation to implementation, however, typically:
+
+1. They store information on the states all children 
+2. They behave like a state machine
+3. They offer some public API to trigger changing the state of children
```
Comment (KovalevAndrey) [inline documentation/navmodel/index.md:13]:
```suggestion
3. They offer some public API to trigger children's state change 
```

File: documentation/navmodel/index.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# Navigation models
+
+Navigation models are one of the most important abstractions of Appyx.
+
+## What's a NavModel?
+
+Navigation models describe navigation itself – by the states and operations they define, any custom navigation mechanism can be implemented. The state is then yours to be controlled from your business logic, which then becomes the state of navigation directly.
```
Comment (KovalevAndrey) [inline documentation/navmodel/index.md:7]:
```suggestion
Navigation models describe navigation with two parameters: Routings (Navigation targets) and State (finite states for Navigation targets). You define which operations your Navigation models support and how they change the state of Routings. 
```

**Comment by LachlanMcKee** [comment]
Can you update the term routing source in the following places?

image

File: core/src/main/java/com/bumble/appyx/core/navigation/RoutingKey.kt [hunk 4]
```diff
@@ -1,4 +1,4 @@
-package com.bumble.appyx.core.routing
+package com.bumble.appyx.core.navigation
```
Comment (LachlanMcKee) [inline core/src/main/java/com/bumble/appyx/core/navigation/RoutingKey.kt:1]: Rename RoutingKey to something else, as it no longer makes sense. `NavKey`/`NavigationKey`

Comment (LachlanMcKee) [reply core/src/main/java/com/bumble/appyx/core/navigation/RoutingKey.kt:1]: also, rename any variables called `routingKey` to `navKey` (or just `key` if posssible)

**Review by LachlanMcKee — CHANGES_REQUESTED** [review]

File: core/src/main/java/com/bumble/appyx/core/navigation/RoutingKey.kt (hunk 4 above)
Comment (LachlanMcKee) [reply core/src/main/java/com/bumble/appyx/core/navigation/RoutingKey.kt:1]: You'll also most likely need to do a second pass of updates in the documentation

Comment (zsoltk) [reply core/src/main/java/com/bumble/appyx/core/navigation/RoutingKey.kt:1]: Maybe a separate PR, it's a large changeset in itself.

Comment (LachlanMcKee) [reply core/src/main/java/com/bumble/appyx/core/navigation/RoutingKey.kt:1]: happy if you want to do it in a separate PR, however I would suggest avoiding renaming them in two different releases.

File: documentation/navmodel/index.md (hunk 3 above)
Comment (zsoltk) [reply documentation/navmodel/index.md:7]: The original phrasing has some key elements I don't want to discard. We can discuss your additions separately, documentation will need more update.

**Comment by LachlanMcKee** [comment]
There are still some renames required within:

`quick-start.md`

> class RootNode(
>     buildContext: BuildContext
> ) : ParentNode<Routing>(
>     routingSource = TODO("We will come back to this in Step 4"),
>     buildContext = buildContext
> ) {

`samples/navigation-compose/README.md`

> This example may be useful if:
> - You want to to use some `RoutingSources` in your non-Appyx app
> - You want to migrate to Appyx gradually

**Review by LachlanMcKee — APPROVED** [review]

**Pull request merged**